git: 2.49.0 -> 2.50.0 + add self as maintainer

me-and opened the pull request:
This also removes adding one of the Bash Git completion scripts in the postInstall phase, as it's now set up automatically by the upstream installer.

I've also added myself as a maintainer, as I'm keen to keep Git up-to-date, and I'm running CI testing of the latest Git branches on Nix at https://github.com/me-and/nix-git-ci

I don't think there's anything worthy of inclusion in the Git release note; the upstream release notes are at https://git.kernel.org/pub/scm/git/git.git/tree/Documentation/RelNotes/2.50.0.adoc?id=f1ca98f609f9a730b9accf24e5558a10a0b41b6c

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built `git` on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested on x86_64-linux:
  - [x] NixOS test `nixosTests.buildbot`
  - [x] Nix tests `tests.fetchgit`
- [x] Tested basic functionality of `gitMinimal`'s `/bin/git`
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

me-and committed:
git: add me-and as maintainer

me-and committed:
git: 2.49.0 -> 2.50.0

This also removes adding one of the Bash Git completion scripts in the
postInstall phase, as it's now set up automatically by the upstream
installer.

Release notes: https://git.kernel.org/pub/scm/git/git.git/tree/Documentation/RelNotes/2.50.0.adoc?id=f1ca98f609f9a730b9accf24e5558a10a0b41b6c

philiptaron reviewed (approved):
Excellent choice. Can you add me too?

me-and committed:
git: add philiptaron as a maintainer

As requested at
https://github.com/NixOS/nixpkgs/pull/417515#pullrequestreview-2939689572

me-and commented:
@philiptaron done at https://github.com/NixOS/nixpkgs/pull/417515/commits/b27ce01ca7a14b6863ca5535c79ae5247fe5a256 :)
